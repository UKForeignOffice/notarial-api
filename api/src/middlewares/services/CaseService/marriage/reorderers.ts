import * as affirmationMappings from "./mappings/affirmation";
import * as cniPostalMappings from "./mappings/cniPostal";
import * as exchangeMappings from "./mappings/exchange";
import * as mscMappings from "./mappings/msc";
import * as cniMSCCombinedMappings from "./mappings/CNIAndMSC";
import { reorderSectionsWithNewName } from "../utils/reorderSectionsWithNewName";

export const reorderers = {
  affirmation: reorderSectionsWithNewName(affirmationMappings.order),
  cni: reorderSectionsWithNewName(affirmationMappings.order),
  cniPostal: reorderSectionsWithNewName(cniPostalMappings.order),
  exchange: reorderSectionsWithNewName(exchangeMappings.order),
  mscPostal: reorderSectionsWithNewName(mscMappings.order),
  cniAndMscPostal: reorderSectionsWithNewName(cniMSCCombinedMappings.order),
};
