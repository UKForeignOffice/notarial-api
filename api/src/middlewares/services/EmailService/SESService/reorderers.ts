import * as affirmationMappings from "./mappings/affirmation";
import { reorderSectionsWithNewName } from "./utils/reorderSectionsWithNewName";
export const reorderers = {
  affirmation: reorderSectionsWithNewName(affirmationMappings.order),
};
