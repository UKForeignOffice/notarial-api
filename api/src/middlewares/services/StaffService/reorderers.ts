import * as affirmationMappings from "./mappings/affirmation";
import * as cniPostalMappings from "./mappings/cniPostal";
import * as exchangeMappings from "./mappings/exchange";
import { reorderSectionsWithNewName } from "./utils/reorderSectionsWithNewName";
export const reorderers = {
  affirmation: reorderSectionsWithNewName(affirmationMappings.order),
  cni: reorderSectionsWithNewName(affirmationMappings.order),
  cniPostal: reorderSectionsWithNewName(cniPostalMappings.order),
  exchange: reorderSectionsWithNewName(exchangeMappings.order),
};
