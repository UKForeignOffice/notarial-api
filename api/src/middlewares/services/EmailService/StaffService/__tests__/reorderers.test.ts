import { remapped } from "./fixtures/fields";
import { reorderers } from "../reorderers";

test("affirmation is successfully remapped", () => {
  const reordered = reorderers.affirmation(remapped);
  expect(reordered).toMatchSnapshot();
});
