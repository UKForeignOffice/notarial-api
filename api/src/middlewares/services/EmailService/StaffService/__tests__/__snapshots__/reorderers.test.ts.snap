// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`affirmation is successfully remapped 1`] = `
{
  "Applicant's address": {
    "addressLine1": {
      "answer": "Test",
      "category": "applicantDetails",
      "key": "addressLine1",
      "title": "Address line 1",
      "type": "text",
    },
    "addressLine2": {
      "answer": "Test",
      "category": "applicantDetails",
      "key": "addressLine2",
      "title": "Address line 2 ",
      "type": "text",
    },
    "city": {
      "answer": "Test",
      "category": "applicantDetails",
      "key": "city",
      "title": "Town or city",
      "type": "text",
    },
    "country": {
      "answer": "United Kingdom",
      "category": "applicantDetails",
      "key": "countryAddress",
      "title": "Country",
      "type": "list",
    },
    "postcode": {
      "answer": null,
      "category": "applicantDetails",
      "key": "postcode",
      "title": "Postcode or zip code",
      "type": "text",
    },
  },
  "Applicant's birth details": {
    "countryOfBirth": {
      "answer": "United Kingdom",
      "category": "applicantDetails",
      "key": "countryOfBirth",
      "title": "Country of birth",
      "type": "list",
    },
    "nameChanged": {
      "answer": "By deed poll",
      "category": "nameChange",
      "key": "nameChanged",
      "title": "Name changed",
      "type": "list",
    },
    "placeOfBirth": {
      "answer": "Test",
      "category": "applicantDetails",
      "key": "placeOfBirth",
      "title": "Place of birth ",
      "type": "text",
    },
  },
  "Applicant's details": {
    "dateOfBirth": {
      "answer": "2000-11-11",
      "category": "applicantDetails",
      "key": "dateOfBirth",
      "title": "Date of birth",
      "type": "date",
    },
    "emailAddress": {
      "answer": "pye@cautionyourblast.com",
      "category": "contact",
      "key": "emailAddress",
      "title": "Email address",
      "type": "text",
    },
    "firstName": {
      "answer": "Test",
      "category": "applicantDetails",
      "key": "firstName",
      "title": "First name",
      "type": "text",
    },
    "middleName": {
      "answer": null,
      "category": "applicantDetails",
      "key": "middleName",
      "title": "Middle name",
      "type": "text",
    },
    "phoneNumber": {
      "answer": "+440000000",
      "category": "contact",
      "key": "phoneNumber",
      "title": "Phone number",
      "type": "text",
    },
    "sex": {
      "answer": "Female",
      "category": "applicantDetails",
      "key": "sex",
      "title": "Male or female",
      "type": "list",
    },
    "surname": {
      "answer": "Test",
      "category": "applicantDetails",
      "key": "surname",
      "title": "Surname",
      "type": "text",
    },
  },
  "Applicant's passport": {
    "dateOfIssue": {
      "answer": "2020-12-12",
      "category": "applicantDetails",
      "key": "passportDateOfIssue",
      "title": "Date of issue of passport",
      "type": "date",
    },
    "number": {
      "answer": "0000",
      "category": "applicantDetails",
      "key": "passportNumber",
      "title": "Passport number",
      "type": "text",
    },
  },
  "Feedback consent": {
    "feedbackConsent": {
      "answer": true,
      "category": "feedback",
      "key": "feedbackConsent",
      "title": "Can our partner contact you for feedback to help improve this service?",
      "type": "list",
    },
  },
  "Marital and civil status": {
    "dateOfDecreeAbsolute": {
      "answer": "2020-12-12",
      "category": "maritalStatus",
      "key": "dateOfDecreeAbsolute",
      "title": "Date of issue of decree absolute",
      "type": "date",
    },
    "placeOfDecreeAbsolute": {
      "answer": "London, England",
      "category": "maritalStatus",
      "key": "placeOfDecreeAbsolute",
      "title": "Place of issue of decree absolute",
      "type": "text",
    },
    "status": {
      "answer": "Divorced",
      "category": "maritalStatus",
      "key": "maritalStatus",
      "title": "Marital status",
      "type": "list",
    },
  },
  "Marriage": {
    "dateOfMarriage": {
      "answer": "2024-06",
      "category": "marriage",
      "key": "dateOfMarriage",
      "title": "Date of marriage",
      "type": "monthYear",
    },
    "placeOfMarriage": {
      "answer": "Test",
      "category": "marriage",
      "key": "placeOfMarriage",
      "title": "Place of marriage",
      "type": "text",
    },
  },
  "Occupation": {
    "occupation": {
      "answer": "Test",
      "category": "occupation",
      "key": "occupation",
      "title": "Occupation",
      "type": "text",
    },
  },
  "Parents' details": {
    "fatherFullName": {
      "answer": "Test Test",
      "category": "parents",
      "key": "fatherFullName",
      "title": "Father's full name",
      "type": "text",
    },
    "motherFullName": {
      "answer": "Test Test",
      "category": "parents",
      "key": "motherFullName",
      "title": "Mother's full name",
      "type": "text",
    },
  },
  "Partner's address": {
    "partnerAddressCity": {
      "answer": "Test",
      "category": "partner",
      "key": "partnerAddressCity",
      "title": "Town or city",
      "type": "text",
    },
    "partnerAddressCountry": {
      "answer": "Spain",
      "category": "partner",
      "key": "partnerAddressCountry",
      "title": "Country",
      "type": "list",
    },
    "partnerAddressLine1": {
      "answer": "Test",
      "category": "partner",
      "key": "partnerAddressLine1",
      "title": "Address line 1",
      "type": "text",
    },
    "partnerAddressLine2": {
      "answer": null,
      "category": "partner",
      "key": "partnerAddressLine2",
      "title": "Address line 2",
      "type": "text",
    },
    "partnerAddressPostcode": {
      "answer": null,
      "category": "partner",
      "key": "partnerAddressPostcode",
      "title": "Postcode or zip code",
      "type": "text",
    },
  },
  "Partner's name": {
    "partnerFirstName": {
      "answer": "Test",
      "category": "partner",
      "key": "partnerFirstName",
      "title": "Partner's first name",
      "type": "text",
    },
    "partnerMiddleName": {
      "answer": null,
      "category": "partner",
      "key": "partnerMiddleName",
      "title": "Partner's middle name",
      "type": "text",
    },
    "partnerSurname": {
      "answer": "Test",
      "category": "partner",
      "key": "partnerSurname",
      "title": "Partner's surname",
      "type": "text",
    },
  },
  "Partner's nationality": {
    "partnerNationality": {
      "answer": "United States citizen",
      "category": "partner",
      "key": "partnerNationality",
      "title": "Partner's nationality",
      "type": "list",
    },
  },
}
`;
